lib/vmpooler/pool_manager.rb

```diff
@@ -258,6 +263,7 @@ def clone_vm(template, folder, datastore, target)
         $redis.decr('vmpooler__tasks__clone')
 
         begin
+          $statsd.timing($config[:statsd]['prefix'] + ".clone.#{vm['template']}", finish) if defined? $statsd
           $graphite.log($config[:graphite]['prefix'] + ".clone.#{vm['template']}", finish) if defined? $graphite
```

Comment:
It looks like you're checking on `$statsd` and `$graphite` a lot. Consider letting the initializer define both of those values (and letting one fall through to `nil`, and then wrapping those into a general method (this is by no means complete but if you're interested then I can help you flesh it out):

```ruby
# @return [SomeKindOfObject, nil] an object representing the defined processing system for metric collection
def metric_collector
  $statsd || $graphite
end
```